hera_cal/nucal.py

```diff
@@ -621,4 +684,281 @@ def build_nucal_wgts(data_flags, data_nsamples, autocorrs, auto_flags, radial_re
+    
+    Parameters:
+    -----------
+    data : dictionary, DataContainer
+        Data to be fit by u-model. This function assumes that visibilities redundantly-averaged
+        and will only use data from the first baseline in a spatially-redundant group.
+    data_wgts : dictionary, DataContainer
+        Weights associated with data
+    radial_reds : RadialRedundancy, list of lists of tuples
+        List of lists of redundant baseline tuples. Each list of redundant tuples represents a radial group of
+        baselines.
+    spatial_filters : dict
+        Dictionary mapping baseline tuples to spatial filters. Dictionary is of the form {(ant1, ant2, pol):
+        np.array([Nfreqs, Nspat])} where Nspat is the number of spatial filters.
+    spectral_filters : np.ndarray, optional, Default is None.
+        Array of DPSS filters with shape (Nfreqs, Nspec) where Nspec is the number of spectral DPSS filters. If None,
+        the model is assumed to be restricted to the spatial axis with no spectral variation.
+    solver : str, optional, Default is 'lu_solve'
+        Solver to use for linear least-squares fit. Options are 'lu_solve', 'solve', 'pinv', and 'lstsq'.
+    tol : float, optional, Default is 1e-15.
+        Regularization tolerance for linear least-squares fit. 
+    share_fg_model : bool, optional, Default is False.
+        If True, the foreground model for each radially-redundant group is shared across the time axis. 
+        Otherwise, a nucal foreground will be independently computed for each frequency.
+    return_model_comps: bool, optional, Default is False.
+        If True, the model components for the foreground model are returned. Otherwise, only the model visibilities
+        are returned. 
+    
+    Returns:
+    --------
+    if return_model_comps:
+        model_comps : list of np.ndarray
+            List of model components projected on to the spectral axis. Each component is a 2D array with shape
+            (Ntimes, Nspec, Nspat) where Ntimes is the number of times in the data, Nspec is the number of DPSS
+            eigenvectors along the spectral axis, and Nspat is the number of spatial filters.
+    else:
+        model : dictionary
+            Dictionary mapping baseline tuples to model visibilities. Dictionary is of the form
+            {(ant1, ant2, pol): np.array([Ntimes, Nfreqs])}.
+    """        
+    # Create empty dictionary for model components
+    model_comps = {}
+
+    # Loop over radial groups
+    for group in radial_reds:
+        # Get the data and design matrix for this group
+        design_matrix = np.array([spatial_filters[bl] for bl in group])
+        wgts_here = np.array([data_wgts[bl] for bl in group])
+        data_here = np.array([data[bl] for bl in group])
+                
+        # Get number of fit parameters
+        if spectral_filters is not None:
+            ndim = spectral_filters.shape[1] * design_matrix.shape[-1]
```

Comment:
I'd add a comment explaining which dimensions these indices are

Reply:
Added to the comment above